src/Akeneo/Bundle/ElasticsearchBundle/Cursor/FromBoundedCursorFactory.php

```diff
@@ -64,24 +63,51 @@ public function __construct(
      */
     public function createCursor($queryBuilder, array $options = [])
     {
+        $options = $this->resolveOptions($options);
+
         $repository = $this->om->getRepository($this->entityClassName);
         if (!$repository instanceof CursorableRepositoryInterface) {
             throw InvalidObjectException::objectExpected($this->entityClassName, CursorableRepositoryInterface::class);
         }
 
-        $pageSize = !isset($options['page_size']) ? $this->pageSize : $options['page_size'];
-        $searchAfter = !isset($options['search_after']) ? [] : $options['search_after'];
-        $searchAfterUniqueKey = !isset($options['search_after_unique_key']) ? null : $options['search_after_unique_key'];
-
         return new $this->cursorClassName(
             $this->searchEngine,
             $repository,
             $queryBuilder,
-            $searchAfter,
             $this->indexType,
-            $pageSize,
+            $options['page_size'],
             $options['limit'],
-            $searchAfterUniqueKey
+            $options['from']
         );
     }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    protected function resolveOptions(array $options)
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefined(
+            [
+                'page_size',
```

Comment:
Not a comment but more a question, why we don't have a CursorWords final class with all those words "page_size", "from", "limit" as we use them in a lot of lines ? It could avoid some mistake and we could see all possible words ? :)